packages/grafana-ui/src/components/QueryField/QueryField.tsx

```diff
@@ -95,7 +95,13 @@ export class QueryField extends React.PureComponent<QueryFieldProps, QueryFieldS
   }
 
   componentDidUpdate(prevProps: QueryFieldProps, prevState: QueryFieldState) {
-    const { query, syntax } = this.props;
+    const { query, syntax, syntaxLoaded } = this.props;
+
+    if (!prevProps.syntaxLoaded && syntaxLoaded && this.editor) {
+      // Need a bogus edit to re-render the editor after syntax has fully loaded
+      const editor = this.editor.insertText(' ').deleteBackward(1);
+      this.onChange(editor.value, true);
```

Comment:
Do we really need to call `this.onChange` here or do we just need to trigger the bogus edit? wdyt @davkal @aocenas @kaydelaney

Reply:
Honestly no idea, moving this to componentDidUpdate may have changed the need for this. @davkal do you know what was this solving/fixing?